tjmperSup/tjmper.db

```diff
@@ -0,0 +1,174 @@
+record(bo, "$(P)SIM")
+{
+    field(SCAN, "Passive")
+    field(DTYP, "Soft Channel")
+    field(ZNAM, "NO")
+    field(ONAM, "YES")
+    field(VAL, "$(RECSIM=0)")
+    field(PINI, "YES")
+}
+
+record(bo, "$(P)DISABLE")
+{
+    field(DESC, "Disable comms")
+    field(PINI, "YES")
+    field(VAL, "$(DISABLE=0)")
+    field(OMSL, "supervisory")
+    field(ZNAM, "COMMS ENABLED")
+    field(ONAM, "COMMS DISABLED")
+}
+
+record(longin, "$(P)ID")
+{
+    field(DESC, "The ID of the device")
+    field(SCAN, "1 second")
+    field(DTYP, "stream")
+    field(INP, "@tjmper.proto getStatus($(P)) $(PORT)")
+    info(archive, "VAL")
+}
+
+record(mbbi, "$(P)MODE")
+{
+    field(DESC, "The current mode of operation")
+    field(ZRST, "All out")                  # All home
+    field(ONST, "PLT1 and SMPL engaged")    # Engage Plate 1 and sample
+    field(TWST, "PLT2 and SMPL engaged")    # Engage Plate 2 and sample
+    info(interest, "HIGH")
+    info(archive, "VAL")
+}
+
+record(mbbiDirect, "$(P)LMT")
+{
+    field(DESC, "Details of the blocks’ positions")
+}
+
+record(bi, "$(P)LMT:PLATE1:HOME")
+{
+    field(INP, "$(P)LMT.B5 CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
+
+record(bi, "$(P)LMT:PLATE1:ENGAGED")
+{
+    field(INP, "$(P)LMT.B4 CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
+
+record(bi, "$(P)LMT:PLATE2:HOME")
+{
+    field(INP, "$(P)LMT.B3 CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
+
+record(bi, "$(P)LMT:PLATE2:ENGAGED")
+{
+    field(INP, "$(P)LMT.B2 CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
+
+record(bi, "$(P)LMT:SAMPLE:HOME")
+{
+    field(INP, "$(P)LMT.B1 CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
+
+record(bi, "$(P)LMT:SAMPLE:ENGAGED")
+{
+    field(INP, "$(P)LMT.B0 CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
+
+record(calc, "$(P)PLATE1:MOVING:_CALC")
+{
+    field(INPA, "$(P)LMT:PLATE1:HOME CP MSS")
+    field(INPB, "$(P)LMT:PLATE1:ENGAGED CP MSS")
+    field(CALC, "(!A) && (!B)")
+}
+
+record(bi, "$(P)PLATE1:MOVING")
+{
+    field(INP, "$(P)PLATE1:MOVING:_CALC CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
+
+record(calc, "$(P)PLATE2:MOVING:_CALC")
+{
+    field(INPA, "$(P)LMT:PLATE2:HOME CP MSS")
+    field(INPB, "$(P)LMT:PLATE2:ENGAGED CP MSS")
+    field(CALC, "(!A) && (!B)")
+}
+
+record(bi, "$(P)PLATE2:MOVING")
+{
+    field(INP, "$(P)PLATE2:MOVING:_CALC CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
+
+record(calc, "$(P)SAMPLE:MOVING:_CALC")
+{
+    field(INPA, "$(P)LMT:SAMPLE:HOME CP MSS")
+    field(INPB, "$(P)LMT:SAMPLE:ENGAGED CP MSS")
+    field(CALC, "(!A) && (!B)")
+}
+
+record(bi, "$(P)SAMPLE:MOVING")
+{
+    field(INP, "$(P)SAMPLE:MOVING:_CALC CP MSS")
+    field(ZNAM, "False")
+    field(ONAM, "True")
+}
```

Comment:
I think this can be done without as much duplication with the help of a substitutions and template file - https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Template-Substitutions 

This also means that if your sample PV name changes, for example, you only need to change it in one place